x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -104,6 +110,40 @@ static DeprecationIssue checkUniqueRealmOrders(final Settings settings, final Pl
         );
     }
 
+    static DeprecationIssue checkImplicitlyDisabledNativeRealms(final Settings settings, final PluginsAndModules pluginsAndModules) {
+        final Map<RealmConfig.RealmIdentifier, Settings> realmSettings = RealmSettings.getRealmSettings(settings);
+        if (realmSettings.isEmpty()) {
+            return null;
+        }
+        // If all configured realms are disabled, this equals to no realm is configured. The implicitly behaviour in this case
+        // is to add file and native realms. So we are good here.
+        if (false == realmSettings.entrySet().stream().anyMatch(
+            e -> e.getValue().getAsBoolean(RealmSettings.ENABLED_SETTING_KEY, true))) {
+            return null;
+        }
+        final List<String> implicitlyDisabledNativeRealmTypes =
+            new ArrayList<>(org.elasticsearch.common.collect.List.of(FileRealmSettings.TYPE, NativeRealmSettings.TYPE));
+        realmSettings.keySet().forEach(ri -> implicitlyDisabledNativeRealmTypes.remove(ri.getType()));
+        if (implicitlyDisabledNativeRealmTypes.isEmpty()) {
+            return null;
+        }
+
+        final String details = String.format(
+            Locale.ROOT,
+            "Found implicitly disabled native %s: [%s]. %s disabled because there are other explicitly configured realms." +
```

Comment:
"native" here means "file or native" right?
Do we need a better word?

Reply:
I struggled with this and could not find a better word. I used `native` because that is what used in the `Realms` class. I agree it is confusing. If possible, I would rename "native realm" to something like "index realm", but I think it would be a cognitive challenge for users. I also thought about the word "builtin" but all realms are kinda "builtin", i.e. part of default distribution. The word "internal" is also used for all realms. How about "basic"?

Reply:
I changed all places to use the term "basic".